examples/wasm/rust/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+                .network(network)
+                .create_wallet_no_persist()
+                .map_err(|e| format!("{:?}", e))?,
+        };
+
+        let client = esplora_client::Builder::new(&esplora_url)
+            .max_retries(6)
+            .build_async()
+            .map_err(|e| format!("{:?}", e))?;
+
+        Ok(WalletWrapper {
+            wallet: wallet,
+            client: client,
+        })
+    }
+
+    pub fn load(changeset_str: &str, url: &str, external_descriptor: &str, internal_descriptor: &str) -> JsResult<WalletWrapper> {
+        let changeset_value: Value = serde_json::from_str(changeset_str)?;
+        let changeset: ChangeSet = serde_json::from_value(changeset_value)?;
+
+        let wallet_opt = Wallet::load()
+            .descriptor(KeychainKind::External, Some(external_descriptor.to_string()))
+            .descriptor(KeychainKind::Internal, Some(internal_descriptor.to_string()))
+            .extract_keys()
+            .load_wallet_no_persist(changeset)?;
+
+        let wallet = match wallet_opt {
+            Some(wallet) => wallet,
+            None => return Err(JsError::new("Failed to load wallet, check the changeset")),
+        };
+
+        let client = esplora_client::Builder::new(&url).build_async()?;
+
+        Ok(WalletWrapper { wallet, client })
+    }
+
+    pub async fn scan(&mut self, stop_gap: usize) -> Result<(), String> {
+        let wallet = &mut self.wallet;
+        let client = &self.client;
+
+        let request = wallet.start_full_scan();
+
+        let update = client
+            .full_scan(request, stop_gap, PARALLEL_REQUESTS)
+            .await
+            .map_err(|e| format!("{:?}", e))?;
+
+        let now = (Date::now() / 1000.0) as u64;
+        wallet
+            .apply_update_at(update, Some(now))
+            .map_err(|e| format!("{:?}", e))?;
+
+        Ok(())
+    }
+
+    pub async fn sync(&mut self, parallel_requests: usize) -> JsResult<()> {
+        let request = self.wallet.start_sync_with_revealed_spks();
+        let update = self.client.sync(request, parallel_requests).await?;
+
+        let now = (Date::now() / 1000.0) as u64;
+        self.wallet.apply_update_at(update, Some(now))?;
+
+        Ok(())
+    }
+    // --8<-- [end:wallet]
+
+    // --8<-- [start:utils]
+    pub fn balance(&self) -> u64 {
+        let balance = self.wallet.balance();
+        balance.total().to_sat()
+    }
+
+    pub fn get_new_address(&mut self) -> String {
+        let address = self
+            .wallet
+            .reveal_next_address(KeychainKind::External);
+
+        address.to_string()
+    }
+    // --8<-- [end:utils]
+
+    pub fn peek_address(&mut self, index: u32) -> String {
+        let address = self
+            .wallet
+            .peek_address(KeychainKind::External, index);
+
+        address.to_string()
+    }
+
+    // --8<-- [start:store]
+    pub fn take_staged(&mut self) -> JsResult<String> {
```

Comment:
@darioAnongba see use of serde_json here. note I'm converting to json string in this example rn because I'm just saving to localStorage, but you can pas JsValue json and stringify on js side